### PR TITLE
4.3 Implement get all sprints

### PR DESCRIPTION
**Pull request by Y-floryanovich**
image
image

### CODE REVIEW

File: src/Teams/Controllers/ManageSprintsController.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            List<Sprint> sprints = await GetAllSprints(team_id, new DisplayOptions { });
+
+            if (sprints == null) 
+                return View("ErrorGetAllSprints");
+
+            var teams = await _manageTeamsService.GetMyTeamsAsync();
+            var team = teams.FirstOrDefault(i => i.Id == team_id);
+
+            if (await _accessCheckService.IsOwnerAsync(team_id)) 
+                ViewBag.AddVision = "visible";
```
Comment (alehkahnovich) [inline src/Teams/Controllers/ManageSprintsController.cs:48]: Please do not use ViewBag, use a model instead

File: src/Teams/Controllers/ManageSprintsController.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        [Authorize]
+        public async Task<IActionResult> AllSprints(int team_id)
+        {
+            List<Sprint> sprints = await GetAllSprints(team_id, new DisplayOptions { });
+
+            if (sprints == null) 
+                return View("ErrorGetAllSprints");
+
+            var teams = await _manageTeamsService.GetMyTeamsAsync();
+            var team = teams.FirstOrDefault(i => i.Id == team_id);
```
Comment (alehkahnovich) [inline src/Teams/Controllers/ManageSprintsController.cs:45]: If we need only one team at this point maybe it's a better solution to use/implement this method on service level?

File: src/Teams/Controllers/ManageSprintsController.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        public ManageSprintsController(IManageSprintsService manageSprintsService, IAccessCheckService accessCheckService, IManageTeamsService manageTeamsService)
+        {
+            _manageSprintsService = manageSprintsService;
+            _manageTeamsService = manageTeamsService;
+            _accessCheckService = accessCheckService;
+        }
+
+        [Authorize, NonAction]
+        public async Task<List<Sprint>> GetAllSprints(int team_id, DisplayOptions options)
```
Comment (alehkahnovich) [inline src/Teams/Controllers/ManageSprintsController.cs:27]: Is there any reason why this action is public are we calling it from the client? If no, then it should be a private method or more preferable to hide this logic inside service layer

**Review by alehkahnovich — CHANGES_REQUESTED** [review]

File: src/Teams/Controllers/ManageSprintsController.cs (hunk 1 above)
Comment (Y-floryanovich) [reply src/Teams/Controllers/ManageSprintsController.cs:48]: Should I use a model, if I need to show only name of team?

File: src/Teams/Controllers/ManageSprintsController.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+        [Authorize]
+        public async Task<IActionResult> AllSprints(int team_id, DisplayOptions options)
+        {
+            List<Sprint> sprints;
+            if (await _accessCheckService.OwnerOrMemberAsync(team_id))
+            {
+                sprints = (List<Sprint>)await _manageSprintsService.GetAllSprintsAsync(team_id, options);
+            }
+            else
```
Comment (SergeyMokin) [inline src/Teams/Controllers/ManageSprintsController.cs:34]: Just an opinion, but prefer to use if-else statement without brackets only in one line

**Review by SergeyMokin — APPROVED** [review]

**Pull request merged**